ZuluContent/Spells/Fifth/PoisonField.cs

```diff
@@ -1,111 +1,120 @@
-                var ry = (dx + dy) * 44;
+            var dx = Caster.Location.X - point.X;
+            var dy = Caster.Location.Y - point.Y;
+            var rx = (dx - dy) * 44;
+            var ry = (dx + dy) * 44;
 
-                bool eastToWest;
+            bool eastToWest;
 
-                if (rx >= 0 && ry >= 0)
-                    eastToWest = false;
-                else if (rx >= 0)
-                    eastToWest = true;
-                else if (ry >= 0)
-                    eastToWest = true;
-                else
-                    eastToWest = false;
+            if (rx >= 0 && ry >= 0)
+                eastToWest = false;
+            else if (rx >= 0)
+                eastToWest = true;
+            else if (ry >= 0)
+                eastToWest = true;
+            else
+                eastToWest = false;
 
-                Effects.PlaySound((Point3D)p, Caster.Map, 0x20B);
+            Effects.PlaySound(point, Caster.Map, 0x20B);
 
-                var itemID = eastToWest ? 0x3915 : 0x3922;
+            var itemId = eastToWest ? 0x3915 : 0x3922;
 
-                var duration = TimeSpan.FromSeconds(3 + Caster.Skills.Magery.Fixed / 25);
+            var power = 2.0;
+            Caster.FireHook(h => h.OnModifyWithMagicEfficiency(Caster, ref power));
 
-                for (var i = -2; i <= 2; ++i)
-                {
-                    var loc = new Point3D(eastToWest ? p.X + i : p.X, eastToWest ? p.Y : p.Y + i, p.Z);
+            var seconds = Caster.Skills[SkillName.Magery].Value / 5.0 + 20.0;
+            Caster.FireHook(h => h.OnModifyWithMagicEfficiency(Caster, ref seconds));
+            var duration = TimeSpan.FromSeconds(seconds);
 
-                    new InternalItem(itemID, loc, Caster, Caster.Map, duration, i);
-                }
+            for (var i = -2; i <= 2; ++i)
+            {
+                var loc = new Point3D(eastToWest ? point.X + i : point.X, eastToWest ? point.Y : point.Y + i, point.Z);
+                var item = new PoisonFieldItem(itemId, loc, Caster, (int)power, duration, i);
             }
+        }
 
-            FinishSequence();
+        private static void ApplyPoison(Mobile caster, Mobile target, int power)
+        {
+            if (caster == null)
+                return;
+            
+            caster.DoHarmful(target);
+
+            var level = SpellHelper.TryResist(caster, target, SpellCircle.Fifth) 
+                ? Utility.Dice(1, (uint)power, 0) 
+                : power;
+
+            
+            var p = Poison.GetPoison(Math.Min(level, Poison.Poisons.Count - 1));
+            target.PlaySound(0x474);
+            
+            if (target.ApplyPoison(caster, p) == ApplyPoisonResult.Poisoned)
+                if (SpellHelper.CanRevealCaster(target))
+                    caster.RevealingAction();
         }
 
         [DispellableField]
-        public class InternalItem : Item
+        public class PoisonFieldItem : Item
```

Comment:
This is almost a carbon copy of the FireField item. For now I'll leave them separate but after I do the up to Energy Field I might see if it's possible to abstract it into a common item all x Field spells can use.

Reply:
yea makes sense